docs/DeveloperGuide.md

```diff
@@ -479,8 +485,68 @@ Given below are instructions to test the app manually.
     2. Test case: `del -m 0` where there is no module `0` in the module list.\
     Expected: As there is no such module in the module list, the deletion will give an error. 
     Details of the associated error message will be shown.
+
+### Editing a task
+1. Editing a task 
+    1. Test case: `edit -t 3 read a book`, where task index `3` has already been added previously into the task list.\
+    Expected: As the task at index `3` is in the task list, the task description will be edited and changed to `read a book`. 
+    Details of the success of the edit task will be shown.
+    2. Test case: `edit -t 10 meet girlfriend`, where there is no task `10` in the task list.\
+    Expected: As there is no such task in the task list, an error will be given to the user. Details of the associated error 
+    message will be shown.
+
+### Editing a module
+1. Editing a module 
+    1. Test case: `edit -m CG2271 GER1000`, where module `CG2271` has already been added previously into the module list.\
+    Expected: As `CG2271` is in the module list, the module will be edited and changed to `GER1000`. Details of the success 
+    of the edit module will be shown.
+    2. Test case: `edit -m ACC1101 GER1000`, where there is no module `ACC1101` in the module list.\
+    Expected: As there is no such module in the module list, an error will be given to the user. Details of the associated 
+    error message will be shown. 
+    
+### Marking a task as done / undone
+1. Marking a task as done
+    1. Test case: `done 1`, where `task` is index `0` in the task list.\
+    Expected: The DoneCommandParser parses `1` and converts it to index `0` in the task list. 
+    As task `task` is the index `0` in the task list, `task` will be marked as done.
+    2. Test case: `done 10`, where there is no task of index `9` in the task list.\
+    Expected: The DoneCommandParser parses `10` and converts it to index `9` in the task list. 
+    As there is no task of index `9` in the task list, an error is thrown. Details of the associated error message will be shown.
+  
+### Viewing the task list
+1. Viewing the task list
+    1. Test case: `list -t`, after adding the tasks "read book", "buy stuff" have been added in succession.\
```

Comment:
Maybe instead of saying add, say that these tasks are in the task list, because it might be possible the task list is already in the json